core/src/org/javarosa/core/model/instance/TreeReference.java

```diff
@@ -337,18 +442,19 @@ else if (!this.getInstanceName().equals(contextRef.getInstanceName())) {
         TreeReference newRef = anchor(contextRef);
         newRef.setContext(contextRef.getContext());
 
-        //apply multiplicites and fill in wildcards as necessary based on the context ref
+        // apply multiplicites and fill in wildcards as necessary, based on the
+        // context ref
         for (int i = 0; i < contextRef.size() && i < newRef.size(); i++) {
-
-            //If the the contextRef can provide a definition for a wildcard, do so
-            if (TreeReference.NAME_WILDCARD.equals(newRef.getName(i)) && !TreeReference.NAME_WILDCARD.equals(contextRef.getName(i))) {
+            // If the the contextRef can provide a definition for a wildcard, do so
+            if (TreeReference.NAME_WILDCARD.equals(newRef.getName(i)) &&
+                    !TreeReference.NAME_WILDCARD.equals(contextRef.getName(i))) {
                 newRef.data.setElementAt(newRef.data.elementAt(i).setName(contextRef.getName(i)), i);
```

Comment:
I have no idea what this code is accomplishing.
As far as I can tell, wildcards in 'this.anchor(contextRef)' will either be present in contextRef too or out of scope, but shouldn't be resolved.